pkg/nodeaddress/node_address.go

```diff
@@ -119,7 +119,19 @@ func InitDefaultPrefix(device string) {
 	ipv6Address = findIPv6NodeAddr()
 
 	ip, err := firstGlobalV4Addr(device)
-	if err == nil {
+	if err != nil {
+		return
+	}
+	// If the IPv6AllocRange is not nil then the allocation should be based
```

Comment:
... then the IPv4 allocation should be derived from the IPv6AllocRange.

Reply:
Fixed